server/src/gameplay/game_event_handler.rs

```diff
@@ -514,6 +514,20 @@ pub fn update_game_state(game_state: &mut GameState, event: Event) -> Result<(),
 
             p.technologies.remove(&tech);
         }
+        Event::SetPlanetOwner { player, planet } => {
+            // Remove the planet from any player that owns it.
+            game_state.players.values_mut().for_each(|p| {
+                p.planets.remove(&planet);
+            });
+
+            if let Some(p) = player {
+                let Some(p) = game_state.players.get_mut(&p) else {
+                    bail!("Player does not exist?");
```

Comment:
this check should ideally happen before game_state is mutated

Reply:
I had it like that at first but game_state borrowing became a nightmare :/

Reply:
Aight, should be fixed now.